primitives/src/merkle_tree/internal.rs

```diff
@@ -212,7 +210,7 @@ where
             .map(|chunk| {
                 let children = chunk
                     .map(|(pos, elem)| {
-                        let pos = I::from(pos as u64);
+                        let pos = pos as u64;
```

Comment:
Nit. `as` could cause bugs. You could throw an error if `usize`->`u64` conversion fails. (It never will because `usize` is at most 8 bytes... for now.) It's convenient that this method already returns a `Result`, so you don't need to obfuscate your API merely to accommodate shortfalls in Rust.

A bug can only occur if you're on a 32-bit system and someone attempts to create a merkle tree with 4 billion elements. So don't lose sleep over this. 😁

Reply:
Nah, the error happens if I'm on a (>64)-bit system and throwing >2^64 elements to create a MT.

Reply:
lol oops! I still blame Rust. 😊